src/transactions/feed/TransactionFeed.tsx

```diff
@@ -36,16 +36,37 @@ function TransactionFeed() {
   const allowedNetworks = getAllowedNetworkIds()
 
   const confirmedFeedTransactions = useMemo(() => {
+    // Filter out received pending transactions that are also in the pending
+    // standby array because those will be displayed with the pending
+    // transactions
+    const completedTransactionsOrWithoutPendingStandby = transactions.filter(
+      (tx) =>
+        tx.status === TransactionStatus.Complete ||
+        !allPendingTransactions.find(
```

Comment:
how long this list may be? if it's long perhaps we can create a Map to reduce the computational complexity

Reply:
in most cases it should be short (this is only the in-flight transactions, not including the completed and failed ones). since users need to make these transactions in the wallet itself to get the standby transactions, it seems reasonable to assume that there is unlikely to be a big buildup of pending transactions especially when we have the internal tx status watcher.